components/chart/canvas/index.tsx

```diff
@@ -34,26 +34,34 @@ function Canvas({
   organizeModules,
   moveModule,
 }: CanvasProps) {
-  const canvasRef = useRef();
-  const wrapperRef = useRef(null);
+  const canvasRef = useRef<HTMLDivElement>(null);
+  const wrapperRef = useRef<HTMLDivElement>(null);
   const [isScrolling, setIsScrolling] = useState(false);
-  const [scrollState, setScrollState] = useState<ScrollPositionType>();
+  const [scrollState, setScrollState] = useState<ScrollPositionType>({
+    scrollLeft: 0,
+    scrollTop: 0,
+    clientX: 0,
+    clientY: 0,
+  });
 
   const onMouseUp = useCallback(() => {
+    if (scrollState.scrollLeft < 50) {
```

Comment:
I understood the problem here, you will need to check the difference between scrollState and canvasRef current scrollstate
Something like next

```suggestion
    const canvas = canvasRef.current;
    if ((canvas.scrollLeft + canvas.scrollTop) - (scrollState.scrollLeft + scrollState.scrollTop) < 50) {
```